packages/image_picker/tizen/src/image_picker_tizen_plugin.cc

```diff
@@ -0,0 +1,310 @@
+        });
+
+    registrar->AddPlugin(std::move(plugin));
+  }
+
+  ImagePickerTizenPlugin() {}
+
+  virtual ~ImagePickerTizenPlugin() {}
+
+ private:
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    LOG_DEBUG("method : %s", method_call.method_name().data());
+
+    if (result_) {
+      SendResultWithError("already_active", "Cancelled by a second request");
+      return;
+    }
+
+    result_ = std::move(result);
+    const auto &arguments = *method_call.arguments();
+
+    ImageSource source = ImageSource::GALLERY;
+    if (method_call.method_name().compare("pickImage") == 0) {
+      double width = 0.0, height = 0.0;
+      int quality = 0;
+      if (std::holds_alternative<flutter::EncodableMap>(arguments)) {
+        flutter::EncodableMap values =
+            std::get<flutter::EncodableMap>(arguments);
+        auto s = values[flutter::EncodableValue("source")];
+        if (std::holds_alternative<int>(s)) {
+          source = (ImageSource)std::get<int>(s);
+        }
+        auto w = values[flutter::EncodableValue("maxWidth")];
+        if (std::holds_alternative<double>(w)) {
+          width = std::get<double>(w);
+        }
+        auto h = values[flutter::EncodableValue("maxHeight")];
+        if (std::holds_alternative<double>(h)) {
+          height = std::get<double>(h);
+        }
+        auto q = values[flutter::EncodableValue("imageQuality")];
+        if (std::holds_alternative<int>(q)) {
+          quality = std::get<int>(q);
+        }
+        imageResize_.SetSize((unsigned int)width, (unsigned int)height,
+                             quality);
+
+      } else {
+        SendResultWithError("Invalid arguments");
+        return;
+      }
+
+      if (source == ImageSource::CAMERA) {
+        // TODO: we need to check this feature after webcam is prepared
+        SendResultWithError("Not supported on this device");
+      } else if (source == ImageSource::GALLERY) {
+        CheckPermissionAndPickImage();
+      } else {
+        SendResultWithError("Invalid image source");
+      }
+
+    } else if (method_call.method_name().compare("pickVideo") == 0) {
+      if (std::holds_alternative<flutter::EncodableMap>(arguments)) {
+        flutter::EncodableMap values =
+            std::get<flutter::EncodableMap>(arguments);
+        auto s = values[flutter::EncodableValue("source")];
+        if (std::holds_alternative<int>(s)) {
+          source = (ImageSource)std::get<int>(s);
+        }
+      } else {
+        SendResultWithError("Invalid arguments");
+        return;
+      }
+
+      if (source == ImageSource::CAMERA) {
+        // TODO: we need to check this feature after webcam is prepared
+        SendResultWithError("Not supported on this device");
+      } else if (source == ImageSource::GALLERY) {
+#ifdef TV_PROFILE
```

Comment:
You may remove this condition and just use `CheckPermissionAndPickImage()`.

Reply:
done.